### PR TITLE
Invalid options in cdx_select component

### PR DESCRIPTION
**Pull request by devduarte**
Fixes: #1339

-If the cdx_component rendered is searchable, then we set typed options to be the value of the select and they get validated in the backend. For now, only batches and samples forms are searchable.

Make the select searchable (thus, allow the user to type in the input) is a functional requirement since they may not know all the available options and they can be searched by their description. The fix feels a little hacky to me, since we are forcing the input value but the component rejects invalid and uses the last one valid, which is the behaviour that is causing the issue. 
Another possibility was clear the select whenever a valid option is not selected, but then the user would save an empty value  without even noticing it.

### CODE REVIEW

File: app/assets/javascripts/components/cdx_select.js.jsx [hunk 1]
```diff
@@ -16,21 +16,27 @@ var CdxSelect = React.createClass({
     }
   },
 
+  onInputChange: function (newValue) {
+    this.selectRef.setValue(newValue)
```
Comment (ysbaddaden) [inline app/assets/javascripts/components/cdx_select.js.jsx:20]: I'd suggest to check for `this.props.searchable` in this callback, and to add a comment explaining why we're updating the selectRef value or not.

**Comment by straight-shoota** [comment]
I think I don't quite understand the purpose of [app/views/samples/_form_specimen_role_js.haml](https://github.com/instedd/cdx/pull/1584/files#diff-01706b92862baf74eaa742770b719cf14bb36409e5ee977521160e658445edc9) and why it's removed (or rather why removing it is fine).

**Comment by devduarte** [comment]
> I think I don't quite understand the purpose of [app/views/samples/_form_specimen_role_js.haml](https://github.com/instedd/cdx/pull/1584/files#diff-01706b92862baf74eaa742770b719cf14bb36409e5ee977521160e658445edc9) and why it's removed (or rather why removing it is fine).

That file had an old autocomplete component we were using for that select and since we're not longer using it, that file is not being used anymore.

File: app/assets/javascripts/components/cdx_select.js.jsx (hunk 1 above)
Comment (devduarte) [reply app/assets/javascripts/components/cdx_select.js.jsx:20]: do you mean something like this? 

```  
 if(this.props.searchable){
   this.selectRef.setValue(newValue) 
 }
```

Comment (devduarte) [reply app/assets/javascripts/components/cdx_select.js.jsx:20]: we already check for that property here: `onInputChange={this.props.searchable && this.onInputChange}>`, adding that check would redundant, wouldn't it? Or there is something that I'm missing?

**Review by ysbaddaden — APPROVED** [review]
I reproduced the bug, and added a note about the behavior to explain why we use this hook.

File: app/views/samples/_form.haml [hunk 2]
```diff
@@ -42,7 +42,7 @@
           = f.label :specimen_role
         .col
           - if @can_update
-            = cdx_select form: f, name: :specimen_role, searchable: true, :class => 'input-x-large' do |select|
+            = cdx_select form: f, name: :specimen_role, searchable: true,  :class => 'input-x-large' do |select|
```
Comment (straight-shoota) [inline app/views/samples/_form.haml:45]:
```suggestion
            = cdx_select form: f, name: :specimen_role, searchable: true, :class => 'input-x-large' do |select|
```

**Review by straight-shoota — APPROVED** [review]

**Pull request merged**